tests/test_cli.py

```diff
@@ -0,0 +1,286 @@
+    return "\n".join(ln if not ln.endswith(":") else elided for ln in s.splitlines())
+
+
+# NOTE: for premature exits in argparse, we must catch SystemExit.
+# for ordinary exits in the CLI code, we can simply inspect the return value.
+
+
+def test_no_args():
+    with patch_stdout_stderr() as (out, _), pytest.raises(SystemExit) as exn:
+        main_under_test([])
+
+    assert out.getvalue() == ""
+    assert exn.value.code != 0
+
+
+def test_help():
+    with patch_stdout_stderr() as (out, err), pytest.raises(SystemExit) as exn:
+        main_under_test(["--help"])
+
+    assert "usage: " in out.getvalue()
+    assert err.getvalue() == ""
+    assert exn.value.code == 0
+
+
+@mock.patch.dict(os.environ, {"COLUMNS": "100000"})  # prevent line-wrapping in --help output
+def test_help_matches_recorded_output():
+    with patch_stdout_stderr() as (out, err), pytest.raises(SystemExit):
+        main_under_test(["--help"])
+
+    assert normalise_help_output(out.getvalue()) == normalise_help_output(EXPECTED_HELP_OUTPUT)
+    assert err.getvalue() == ""
+
+
+def test_one_argument_is_status(mock_network_requests):
+    with patch_stdout_stderr() as (out, err):
+        assert main_under_test(["example.com"]) == 0
+
+    assert (
+        "version: Java 1.8-pre1 (protocol 44)\n"
+        "motd: \x1b[0mA Minecraft Server\x1b[0m\n"
+        "players: 0/20 No players online\n"
+        "ping: 0.00 ms\n"
+    ) == out.getvalue()
+    assert err.getvalue() == ""
+
+
+def test_status(mock_network_requests):
+    with patch_stdout_stderr() as (out, err):
+        assert main_under_test(["example.com", "status"]) == 0
+
+    assert (
+        "version: Java 1.8-pre1 (protocol 44)\n"
+        "motd: \x1b[0mA Minecraft Server\x1b[0m\n"
+        "players: 0/20 No players online\n"
+        "ping: 0.00 ms\n"
+    ) == out.getvalue()
+    assert err.getvalue() == ""
+
+
+def test_status_bedrock(mock_network_requests):
+    with patch_stdout_stderr() as (out, err):
+        assert main_under_test(["example.com", "--bedrock", "status"]) == 0
+
+    assert (
+        "version: Bedrock 1.18.100500 (protocol 422)\n"
+        "motd: \x1b[0m\x1b[0m\x1b[0m\x1b[38;2;170;0;0mG\x1b[0m\x1b[0m\x1b[38;2;255;170;0ma\x1b[0m\x1b[0m\x1b[38;2;255;255;85m"
+        "y\x1b[0m\x1b[0m\x1b[38;2;0;170;0mB\x1b[0m\x1b[0m\x1b[38;2;0;0;170mo\x1b[0m\x1b[0m\x1b[38;2;85;85;255mw\x1b[0m\x1b[0m"
+        "\x1b[38;2;255;85;255ms\x1b[0m\x1b[0m\x1b[38;2;170;0;0me\x1b[0m\x1b[0m\x1b[38;2;255;170;0mr\x1b[0m\n"
+        "players: 1/69\n"
+        "ping: 0.00 ms\n"
+    ) == out.getvalue()
+    assert err.getvalue() == ""
+
+
+def test_status_offline(mock_network_requests):
+    with patch_stdout_stderr() as (out, err), patch("mcstatus.server.JavaServer.status", side_effect=socket.timeout):
+        assert main_under_test(["example.com", "status"]) == 1
+
+    assert out.getvalue() == ""
+    assert err.getvalue() == "Error: \n"
```

Comment:
Wait what? `"Error: \n` is all that stderr contains? Shouldn't it follow with the actual error/traceback? I'm not saying to test for that, but shouldn't this be `err.getvalue().startswith("Error: \n")`?

Reply:
No, that's right. It is because in the patch, error doesn't have a message, just `TimeoutError()`. I can change here to repr if you want, but this shouldn't impact real users.
https://github.com/py-mine/mcstatus/blob/3d0c7a2c017fae6cea55bc0c23043609788d4117/mcstatus/__main__.py#L197

Reply:
I see, it should definitely use repr